libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3816,6 +3816,20 @@ void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 				)
 			);
 
+		if (
+			functionDefinition.visibility() == Visibility::Private &&
+			functionDefinition.scope() != m_currentContract
+		)
+			m_errorReporter.typeError(
+				6772_error,
+				path->location(),
```

Comment:
Would be nice to have a secondary location pointing at the function declaration.

```suggestion
				path->location(),
				SecondarySourceLocation().append(
						"Function defined here:",
						functionDefinition.location()
				),
```

Also for the other error above.

And make sure to check the output by manually compiling something on the CLI. The secondary messages do not show up in isoltest output.